Gcpe.Hub.API.Tests/ControllerTests/ActivitiesControllerTests.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoMapper;
+using FluentAssertions;
+using Gcpe.Hub.API.Controllers;
+using Gcpe.Hub.API.Data;
+using Gcpe.Hub.Data.Entity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Gcpe.Hub.API.Tests.ControllerTests
+{
+    public class ActivitiesControllerTests : IDisposable
+    {
+        private Mock<ILogger<ActivitiesController>> logger;
+        private HubDbContext context;
+        private IMapper mapper;
+        private DbContextOptions<HubDbContext> options;
+
+        public ActivitiesControllerTests()
+        {
+            this.options = new DbContextOptionsBuilder<HubDbContext>()
+                      .UseInMemoryDatabase(Guid.NewGuid().ToString())
+                      .Options;
+            this.context = new HubDbContext(this.options);
+
+            this.logger = new Mock<ILogger<ActivitiesController>>();
+
+            var mockMapper = new MapperConfiguration(cfg =>
+            {
+                cfg.AddProfile(new MappingProfile());
+            });
+            this.mapper = mockMapper.CreateMapper();
+        }
+        public void Dispose()
+        {
+            this.context.Dispose();
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(3)]
+        public void GetNewsForecast_ShouldReturnSuccess(int messageCount)
+        {
+            for (var i = 0; i < messageCount; i++)
+            {
+                var dbActivity = TestData.CreateDbActivity(i.ToString(), "test details", i + 1);
+                context.Activity.Add(dbActivity);
+            }
+            context.SaveChanges();
+            var controller = new ActivitiesController(context, logger.Object, mapper);
+
+            var result = controller.GetNewsForecast(7) as ObjectResult;
+
+            result.Should().BeOfType<OkObjectResult>();
+            result.Should().NotBeNull();
+            var models = result.Value as ICollection<Models.Activity>;
+            models.Should().NotBeNull();
+            models.Count().Should().Be(messageCount);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(7)]
+        public void GetNewsForecast_ShouldFilterOnNumDays(int numDays)
```

Comment:
This test should create activities that are beyond the filter number of days and assert that they aren't returned. Currently it doesn't filter anything out.